python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -53,33 +52,37 @@ def __init__(self):
         ret_val_doc += "For the 'update_blocked' status, data will contain a reason key containing an explanation."
 
         self.parameters = {"return_value": {"description": ret_val_doc, "type": "dict" }}
-            
+
     def run_noninteractive(self, log, parameters):
         """
-        Tank command API accessor. 
+        Tank command API accessor.
         Called when someone runs a tank command through the core API.
-        
+
         :param log: std python logger
         :param parameters: dictionary with tank command parameters
         """
-        return self._run(log, True)
-    
+        return self._run(log, True, parameters[0] if len(parameters) else None)
```

Comment:
The non interactive version never cared for the number of arguments, so we won't care if there is more than 1 to make sure we don't break anything.